### PR TITLE
TVDomainGuard factory

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Followup of https://github.com/csarofeen/pytorch/pull/1940

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/test/test_gpu_utils.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #include <gtest/gtest.h>
```
Comment (zasdfgbnm) [inline torch/csrc/jit/codegen/cuda/test/test_gpu_utils.cpp:3]: renaming this file to `test_gpu_utils.cpp`. Tests for testing utilities inside `scheduler_utils`, `ir_utils` should go to this file. `test_gpu.cpp` is already too large, and we should avoid adding new tests to it if possible.

File: torch/csrc/jit/codegen/cuda/lower_utils.cpp [hunk 2]
```diff
@@ -36,13 +36,42 @@ kir::IfThenElse* cloneIfThenElse(kir::IfThenElse* ite) {
...
 namespace ir_utils {
 
-TVDomainGuard::TVDomainGuard(TensorView* _tv, TensorDomain* td)
-    : tv_(_tv), prev_domain(tv_->domain()) {
+TVDomainGuard::TVDomainGuard(TensorView* tv, TensorDomain* td)
+    : tv_(tv), prev_domain_(tv_->domain()) {
   tv_->setDomain(td);
 }
 
+TVDomainGuard::TVDomainGuard(TVDomainGuard&& guard)
```
Comment (shmsong) [inline torch/csrc/jit/codegen/cuda/lower_utils.cpp:44]: nit: should we delete copy constructor?

File: torch/csrc/jit/codegen/cuda/lower_utils.h [hunk 3]
```diff
@@ -39,24 +39,30 @@ namespace ir_utils {
...
   //!  information that can only be reliably kept at the original domain.
   const TensorDomain* prevDomain() const {
-    return prev_domain;
+    return prev_domain_;
   }
 
   ~TVDomainGuard();
 };
 
+TORCH_CUDA_CU_API ir_utils::TVDomainGuard overrideContiguityGuard(
```
Comment (shmsong) [inline torch/csrc/jit/codegen/cuda/lower_utils.h:64]: Comments here would be helpful.

**Review by shmsong — APPROVED** [review]
LGTM. Really nice clean up!

File: torch/csrc/jit/codegen/cuda/lower_utils.cpp (hunk 2 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/lower_utils.cpp:44]: deleted in https://github.com/csarofeen/pytorch/pull/1953/files#diff-5760d90c9f11c9ac3eaeb65d7ab72f4008ada59e0688f05cf77bdd6dc4d28c0dR49

File: torch/csrc/jit/codegen/cuda/lower_utils.h (hunk 3 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/lower_utils.h:64]: added

**Pull request merged**